snapshotter/demux/metrics/proxy.go

```diff
@@ -30,6 +33,75 @@ const (
 	MaxMetricsResponseSize = 32768 // 32 KB
 )
 
+// Monitor is used for port and lifecycle management of metrics proxies.
+type Monitor struct {
+	// monitorChan indicates a port is free for metrics proxy server.
+	ch chan int
+	// portMap tracks used and unused metrics proxy server ports.
+	portMap map[int]bool
+	// portRange is the configurable range of ports to serve metrics proxies from.
+	portRange string
```

Comment:
How about keeping the begin/end of the range, instead of its string representation. This struct doesn't have to know the string representation we have chosen in the TOML file.

Reply:
addressed in recent commit 👍